### ISSUE TITLE
SSH ControlMaster path handling broken on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2574]
This seems to be a core issue underlying many test failures. It forces POSIX path concatenation, instead of os.path.join.

**Comment by mih** [comment]
Still an issue, here the output (git bash) windows 7:

```
 MINGW32 /h/studyforrest-data-aligned (adjusted/master(unlocked))
$ datalad create-sibling this.example.com:dummyrepo
[INFO] Connecting ...
command-line line 0: Missing argument.
[WARNING] Failed to run cmd ['ssh', '-fN', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=15m', '-o', 'ControlPath="C:\\Users\\mihanke\\AppData\\Local\\datalad.org\\datalad\\Cache\\sockets/18f493ff"', 'this.example.com']. Exit code=255
| stdout: None
| stderr: None
[ERROR] Cannot open SSH connection to this.example.com [sshconnector.py:__call__:129] (RuntimeError)
```

**Comment by mih** [comment]
But a switch to switch to `op.join()` doesn't fix the issue. It still fails when executed by datalad. However, when I take the command that is now reported failing, and run it in the git-bash myself, it does what it should.

**Comment by yarikoptic** [comment]
Ran into the same while trying to make `@with_testrepos` work on windows:
![gkrellshoot_09-24-18_145921](https://user-images.githubusercontent.com/39889/45972774-a083d380-c00a-11e8-8354-8e61c8af22c5.png)
for me it stalls the entire process, can't even interrupt in that conda shell... pain

**Comment by yarikoptic** [comment]
ha ha -- but in my case overall it is another issue -- "Detected ssh ri" -- so it just incorrectly determines it to be ssh url which shouldn't be the case here ;-)

**Comment by bpoldrack** [comment]
> but in my case overall it is another issue -- "Detected ssh ri"

It's not another issue, but an additional one. The path considered an SSH RI is not the socket path. That one still fails the same way (not existing because of wrong concatenation).
However, that's interesting. I remember the detection was supposed to first check for a windows path and only then for SSH. We were aware of this potential confusion at some point.

**Comment by mih** [comment]
The release of 0.12 with a fix is imminent (#3993) closing.
